www/js/ui.js

```diff
@@ -618,7 +618,7 @@ $("#shuffleplaylist").click(function() {
 /* load channel */
 
 var loc = document.location+"";
-var m = loc.match(/\/r\/([a-zA-Z0-9-_]+)/);
+var m = loc.match(/\/\w+\/([a-zA-Z0-9-_]+)/);
```

Comment:
That particular snippet dates back to when the URL was first changed from `/?channel=whatever` to `/r/whatever`.  I don't know why it was done that way, it's probably not necessary (although you would still need to check that `document.location.pathname.split('/')[2]` is a valid channel name in your example).